app/models/application_merits_task/appeal.rb

```diff
@@ -1,5 +1,15 @@
 module ApplicationMeritsTask
   class Appeal < ApplicationRecord
     belongs_to :legal_aid_application
+
+    before_save :sync_original_judge_level
+
+  private
+
+    def sync_original_judge_level
+      if second_appeal?
+        self.original_judge_level = nil
```

Comment:
I don't understand this, why set the value to nil if changing second_appeal?
I'm sure there's a good reason, but maybe a comment to clarify it would help future devs?

Reply:
sure, will add. it's because the original judge question is not required for second appeals, only for "first" appeals. So if they change their answer from "No", to "Yes" then we want to clear the now irrelevant original_judge_level

Reply:
yeah, i see other examples in forms. I guess my database centric brain just immediately went for a model callback, of which there are a fair few in the codebase (before_save, after_commit). I guess i should follow the "form" based pattern as it is more prevalent, even if it feels less safe to me - as someone who loves updating data in the console 😄.

Reply:
@colinbruce I have added a commit that does the same thing via a form based approach, with some comments on impact. I am not fussed but while it does adhere to the pattern of similar forms and is less "magical" than callbacks, it has disadvantages too. 

What do you prefer. Happy to go with this regardless, but may be worth a wider discussion going forward (have put a placeholder in dev time trello for now regardless)

Reply:
@colinbruce have added a second alternative which:

1. uses custom callbacks so we don't not have to override default `def save` 
2. uses attribute assignment rather than model update* 

_*which i think would have worked via the `def save` override approach too, but does not allow for "where" to do the assignment except before or after whatever `super` would call_

I rather like the callback approach to help keep form objects small/thin

Reply:
Agreed by developers to introduce this new pattern